base/src/main/java/org/mozilla/jss/provider/javax/crypto/JSSTrustManager.java

```diff
@@ -428,4 +443,26 @@ public X509Certificate[] getAcceptedIssuers() {
 
         return caCerts.toArray(new X509Certificate[caCerts.size()]);
     }
+
+    public void nssCertVerify(X509Certificate[] certChain, String keyUsage) throws CertificateException {
+        try{
+            CryptoManager cm = CryptoManager.getInstance();
+
+            logger.debug("JSSTrustManager: nssCertVerify()");
+            for (int i = 0; i < certChain.length; i++) {
+                CertificateUsage usage = CertificateUsage.AnyCA;
+
+                if (i == certChain.length - 1) {
+                    usage = keyUsage.equals(SERVER_AUTH_OID) ? CertificateUsage.SSLServer : CertificateUsage.SSLClient;
+                }
+
+                cm.verifyCertificate((org.mozilla.jss.crypto.X509Certificate) certChain[i], false, usage);
+            }
+        } catch (NotInitializedException e) {
+            logger.error("JSSTrustManager: Unable to get CryptoManager: " + e, e);
+            throw new RuntimeException(e);
+        } catch (ObjectNotFoundException | InvalidNicknameException ex) {
+            logger.error("JSSTrustManager: nss certificate verification not done: " + ex, ex);
+        }
```

Comment:
Not sure if these exceptions will ever happen, but if it does happen I think we should rethrow the exception (or let it be handled by the caller).